packages/bff/src/services/agreementService.ts

```diff
@@ -1,9 +1,107 @@
 /* eslint-disable functional/immutable-data */
-/* eslint-disable max-params */
-import { agreementApi, catalogApi } from "pagopa-interop-api-clients";
-import { getAllFromPaginated } from "pagopa-interop-commons";
-import { AgreementProcessClient } from "../providers/clientProvider.js";
-import { Headers } from "../utilities/context.js";
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+
+import {
+  getAllFromPaginated,
+  toSetToArray,
+  WithLogger,
+} from "pagopa-interop-commons";
+import {
+  bffApi,
+  catalogApi,
+  agreementApi,
+  tenantApi,
+  attributeRegistryApi,
+} from "pagopa-interop-api-clients";
+import {
+  AgreementProcessClient,
+  PagoPAInteropBeClients,
+} from "../providers/clientProvider.js";
+import { BffAppContext, Headers } from "../utilities/context.js";
+import { agreementDescriptorNotFound } from "../model/domain/errors.js";
+import {
+  toCompactEservice,
+  toCompactDescriptor,
+} from "../model/api/apiConverter.js";
+import { getBulkAttributes } from "./attributeService.js";
+import { enhanceTenantAttributes } from "./tenantService.js";
+
+export function agreementServiceBuilder(clients: PagoPAInteropBeClients) {
+  const { agreementProcessClient } = clients;
+  return {
+    async createAgreement(
+      payload: bffApi.AgreementPayload,
+      { headers, logger }: WithLogger<BffAppContext>
+    ) {
+      logger.info(`Creating agreement with seed ${JSON.stringify(payload)}`);
```

Comment:
please never blindly log the payload.
We should always consider that sensitive data could be added in the future, and it should not be persisted in logs.

If any value is useful, and it is not already logged in the process, it should be added explicitly

Reply:
That is absolutely a good catch, in scala a similar thing has been done: https://github.com/pagopa/interop-be-backend-for-frontend/blob/b37f24cc27d7690cc9ece8dd215a16437de2f0cf/src/main/scala/it/pagopa/interop/backendforfrontend/service/impl/AgreementProcessServiceImpl.scala#L29

In this case, what would be the preferred log behaviour?

Reply:
ouch, thanks for pointing it out!

In this case the most useful details are
- consumer id (the organization that is making the request)
- eservice id
- descriptor id